src/cb_admin/index.js

```diff
@@ -14,19 +11,27 @@ import Feedback from './pages/Feedback';
 import NotFound from '../shared/components/NotFound';
 
 
-export default () => (
-  <Switch>
-    <Route exact path="/cb/password/reset/:token" component={ResetPassword} />
-    <Route exact path="/cb/password/forgot" component={ForgotPassword} />
-    <Route exact path="/cb/login" component={Login} />
-    <Route exact path="/cb/confirm" component={ConfirmPassword} />
-    <Route exact path="/cb/dashboard" component={Dashboard} />
-    <Route exact path="/cb/activities" component={Activities} />
-    <Route exact path="/cb/visitors" component={VisitorDetails} />
-    <Route exact path="/cb/visits" component={VisitsData} />
-    <Route exact path="/cb/visitors/:id" component={Visitor} />
-    <Route exact path="/cb/settings" component={Settings} />
-    <Route exact path="/cb/feedback" component={Feedback} />
-    <Route component={NotFound} />
-  </Switch>
-);
+export default class CbAdminRoutes extends React.Component {
+
+  state = {
+    auth: false,
+  }
+
+  loggedIn = () => this.setState({ auth: true })
+
+  render() {
+    return this.state.auth
+      ? (<Switch>
+        <Route exact path="/admin/activities" component={Activities} />
+        <Route exact path="/admin/visitors" component={VisitorDetails} />
+        <Route exact path="/admin/visits" component={VisitsData} />
+        <Route exact path="/admin/visitors/:id" component={Visitor} />
+        <Route exact path="/admin/settings" component={Settings} />
+        <Route exact path="/admin/feedback" component={Feedback} />
+        <Route exact path="/admin" component={Dashboard} />
+        <Route component={NotFound} />
+      </Switch>)
+      : <ConfirmPassword loggedIn={this.loggedIn} />;
```

Comment:
Since this is essentially an event listener, let's try to stick with the conventional "onXXXX" naming, and call it `onLogin`.